src/player/minimax.rs

```diff
@@ -0,0 +1,253 @@
+                    startmove.id());
+            }
+            return startmove;
+        } else if pre_depth == 1 { // second moves always win (?)
+            let secondmove = Random::new(me).next_move(state);
+            if (me == EdgeState::Blue) {
+                println!("Blue thinks: second move always wins, chooses randomly: {}",
+                    secondmove.id());
+            } else {
+                println!("Red thinks: second move always wins, chooses randomly: {}",
+                    secondmove.id());
+            }
+            return secondmove;
+        }
+
+        // Using this to track our progression through game rounds/tree depth
+        let mut depth = pre_depth;
+
+        // use an array to represent the move sequence (this is basically our tree
+        // thats being expanded)
+        let mut move_sequence: [u8; 15] = [0; 15];
+
+        // using this to clearly represent the move thats currently being considered
+        let mut current_move: u8 = 0;
+
+        // using this to signal that we have just ascended in the tree (not descended)
+        // and thus need to look for the next branch - or ascend further
+        let mut ascend: bool = false;
+
+        // Using this to signal that we want to keep looking for more moves on the current level
+        let mut samelevel: bool = false;
+
+        // This reflects the nature of the minimax problem, switching rounds means the players
+        // change and aim to achieve a minimization/maximization (from the perspective of "me"
+        // player)
+        // In this case, as theres only 2 possible game outcomes (win, lose) we can reduce it
+        // to bools.
+        // "True": downwards a branch, "Me"-Player can force a win
+        // "False": downwards a branch, "Other"-Player can force a win
+        // "Me"-Player estimations will be initialized to "false" and then choose the maximum
+        // of next lowest layer results (downgoing branches).
+        // "Other"-Player estimations will be initialized to "true" and then choose the maximum
+        // of next lowest layer results (downgoing branches).
+        // Can replace minimization by "and"-ing the next lowest layer minimax result into the
+        // current layer
+        // Can replace maximization by "or"-ing the next lowest layer minimax result into the
+        // current layer.
+        // We can skip branches (in an alpha/beta-pruning sort of way) on estimation changes
+        // from true -> false (definite minimum) and from false -> true (definite maximum)
+        let mut minimax: [bool; 15] = [true; 15];
+
+        // Track amount of expanded positions
+        let mut counter: u64 = 0;
+
+        // now descend depth-first through the move sequence tree and let the leaf results
+        // propagate upwards our minimax structure
+        loop {
+            if depth > 14
+            {
+                panic!("How could this happen to me");
+            }
```

Comment:
You can also replace this if with `assert!(depth <= 14, "How could this happen to me");` if you want. But I wouldn't consider one version clearly better than the other one.